### PR TITLE
Optimize debian-iptables image generation

### PR DESCRIPTION
**Pull request by rikatz**
Signed-off-by: Ricardo Katz <rkatz@vmware.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Followup of https://github.com/kubernetes/kubernetes/issues/102493 proceeding with debian-iptables image cleanup

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/kubernetes/issues/102493.

#### Does this PR introduce a user-facing change?

```release-note
- debian-iptables: Build buster-v1.6.3 image
- setcap: Build buster-v2.0.2 image
```

### CODE REVIEW

**Comment by rikatz** [comment]
/cc @justaugustus @BenTheElder  

Final one :) 🤞

File: images/build/debian-iptables/buster/Dockerfile [hunk 1]
```diff
@@ -44,3 +43,6 @@ RUN update-alternatives \
 	--install /usr/sbin/ip6tables ip6tables /usr/sbin/iptables-wrapper 100 \
 	--slave /usr/sbin/ip6tables-restore ip6tables-restore /usr/sbin/iptables-wrapper \
 	--slave /usr/sbin/ip6tables-save ip6tables-save /usr/sbin/iptables-wrapper
+
+FROM scratch
+COPY --from=build / /
```
Comment (BenTheElder) [inline images/build/debian-iptables/buster/Dockerfile:48]: neat! do we know that this reduces the size?

Comment (rikatz) [reply images/build/debian-iptables/buster/Dockerfile:48]: yes! I've tested :)

gcr.io/k8s-staging-build-image/debian-iptables-amd64     buster-v1.6.3                      780ad1ada900   4 minutes ago        60.2MB

vs

gcr.io/k8s-staging-build-image/debian-iptables-amd64     buster-v1.6.2                      830a8bea5d2d   27 hours ago         87.7MB

Per @thockin suggestion on the original issue :D

**Review by BenTheElder — COMMENTED** [review]
`k8s.gcr.io/build-image/debian-base: dependents` needs updating

File: images/build/debian-iptables/buster/Dockerfile (hunk 1 above)
Comment (BenTheElder) [reply images/build/debian-iptables/buster/Dockerfile:48]: If we were still using debian-base for other component images, I might point out that some of that size is probably shared layers that can't be shared anymore, but I think here this is probably a reasonable trade-off (with other runtime components being on debian-iptables or distroless, or something else entirely)

**Comment by rikatz** [comment]
@BenTheElder I'm wondering if the other images generated using Debian-base should also be updated to use Debian-base on its new version (setcap actually)

**Comment by BenTheElder** [comment]
IMHO: yes. You could make upgrading debian-base across the repo an additional (first?) commit and split out the debian-iptables changes to a second commit.

dependencies.yaml should be ensuring we keep things in sync.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Review by justaugustus — APPROVED** [review]
Thanks again, @rikatz! 🎉

**Comment by justaugustus** [comment]
Promotion PR: https://github.com/kubernetes/k8s.io/pull/2199

**Pull request merged**